puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/nightly_packaging.groovy

```diff
@@ -0,0 +1,42 @@
+def setup_nightly_build_environment(args) {
+    def commit_hash = ''
+    def project_name = args.project_name
+    def git_url = args.git_url
+    def git_ref = args.git_ref
+    def ruby_version = args.ruby_version ?: env.ruby_version
+
+    dir(project_name) {
+        // the full checkout() is necessary for checking out specific commits,
+        // as opposed to just branches with git()
+        checkout([
+            $class : 'GitSCM',
+            branches : [[name: git_ref]],
+            extensions: [[$class: 'CleanCheckout']],
+            userRemoteConfigs: [[url: git_url]]
+        ])
+    }
+    dir('foreman-packaging') {
+        git(url: 'https://github.com/theforeman/foreman-packaging.git', branch: 'rpm/develop')
+    }
+    setup_obal()
+    configureRVM(ruby_version)
```

Comment:
Only needed when building source files so I wonder if this shouldn't be part of that function.

Reply:
All sources build in this PR are ruby based (or require a ruby) to build the source.

Reply:
The point I'm making is that I'm not sure a setup function is the right approach. In the UI there's a button that allows you to rerun a stage. Because we split setup and actual run into separate stages, that always fails because then you're missing the setup parts. Perhaps it's a bigger issue and doesn't have to block this PR.

Reply:
Okay. I'd consider that a larger pipeline design issue/discussion because we have *lots* of pipelines with a setup task.

Reply:
Agreed